### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by vc-jinal**
Created a JSON file with food items . It contain Pizza, Burger and Noodles .
Created CRUD for each menu item with routing files.
Used  HTTP Verbs and Resource names.
Store file name in .env file

### CODE REVIEW

File: crud-routing/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.env
+node_modules
+p.js
```
Comment (vc-mohit) [inline crud-routing/.gitignore:3]: p.js is required to put in git ignore?

File: crud-routing/.vscode/settings.json
Comment (vc-mohit) [inline crud-routing/.vscode/settings.json:1]: don't push .vscode folder

File: crud-routing/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "start": "index.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "body-parser": "^1.20.2",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "nodemon": "^2.0.22"
```
Comment (vc-mohit) [inline crud-routing/package.json:16]: nodemon should be in dev dependency

File: crud-routing/.gitignore (hunk 1 above)
Comment (vc-jinal) [reply crud-routing/.gitignore]: removed it

File: crud-routing/.vscode/settings.json
Comment (vc-jinal) [reply crud-routing/.vscode/settings.json:1]: removed .vscode folder

File: crud-routing/package.json (hunk 2 above)
Comment (vc-jinal) [reply crud-routing/package.json:18]: changed nodemon dependencies to dev dependency

**Review by vc-mohit — APPROVED** [review]

File: crud-routing/burger.router.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+const burgerRouter = require('express').Router();
+const bodyParser = require('body-parser');
+const fs = require('fs');
+require('dotenv').config();
+const menuFile = process.env.MENU_FILE;
+
+burgerRouter.use(bodyParser.json());
```
Comment (gaurangviitorcloud) [inline crud-routing/burger.router.js:7]: use body parser once in main file don't need to use it in all routes also body-parses package is not needed i thought we had discussed this.

File: crud-routing/burger.router.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+burgerRouter.get('/', (req, res) => {
+    const menuFile = process.env.MENU_FILE;
+    const data = JSON.parse(fs.readFileSync(menuFile));
+    res.json(data.burger);
+});
+
+// get burger by id
+burgerRouter.get('/:id', (req, res) => {
+    const menuFile = process.env.MENU_FILE;
+    const data = JSON.parse(fs.readFileSync(menuFile, 'utf8'));
```
Comment (gaurangviitorcloud) [inline crud-routing/burger.router.js:19]: use async methods for all file operations

File: crud-routing/burger.router.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    data.burger.push(newBurger);
+    fs.writeFileSync(menuFile, JSON.stringify(data));
+    res.json(newBurger);
+});
+
+// PUT (update) a burger by ID
+burgerRouter.put('/:id', (req, res) => {
+    const menuFile = process.env.MENU_FILE;
+    const data = JSON.parse(fs.readFileSync(menuFile, 'utf8'));
```
Comment (gaurangviitorcloud) [inline crud-routing/burger.router.js:50]: If menu file is a json then JSON.parse is not required please google how to import json file in node js application

File: crud-routing/menu.json [hunk 6]
```diff
@@ -0,0 +1 @@
+{"pizza":[{"id":3,"name":"sssssssssssssssssssssssssss","price":190},{"id":4,"name":"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","price":190},{"id":5,"name":"oooooooooooooooooo","price":190},{"id":6,"name":"Maharaja pizza","price":190}],"burger":[{"id":2,"name":"Burger","price":190}],"noodles":[{"id":1,"name":"chinese noodles","price":190},{"id":2,"name":"n2","price":150},{"id":3,"name":"Hakka noodles","price":250},{"id":4,"name":"n999","price":250}],"counter":[{"pizza":7,"burger":5,"noodles":1}]}
```
Comment (gaurangviitorcloud) [inline crud-routing/menu.json:1]: Why counter is an array ?

**Review by gaurangviitorcloud — CHANGES_REQUESTED** [review]

File: crud-routing/index.js [hunk 7]
```diff
@@ -1,14 +1,12 @@
 const express = require('express');
 const app = express();
 const bodyParser = require('body-parser');
-// const fs = require('fs');
+
+app.use(bodyParser.json())
```
Comment (gaurangviitorcloud) [inline crud-routing/index.js:5]: body-parser package is not required

**Review by gaurangviitorcloud — CHANGES_REQUESTED** [review]
Please read the comments properly if there is a doubt feel free to ask

File: crud-routing/burger.router.js (hunk 3 above)
Comment (vc-jinal) [reply crud-routing/burger.router.js]: remove body-parser

File: crud-routing/burger.router.js (hunk 4 above)
Comment (vc-jinal) [reply crud-routing/burger.router.js]: used async method for all file operations

File: crud-routing/burger.router.js (hunk 5 above)
Comment (vc-jinal) [reply crud-routing/burger.router.js]: remove JSON.parse

File: crud-routing/menu.json (hunk 6 above)
Comment (vc-jinal) [reply crud-routing/menu.json]: remove array

**Review by vc-mohit — APPROVED** [review]

**Review by gaurangviitorcloud — APPROVED** [review]
LGTM

**Pull request merged**